core/src/main/java/io/jenkins/pluginhealth/scoring/probes/JiraOpenIssuesProbe.java

```diff
@@ -0,0 +1,127 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.jenkins.pluginhealth.scoring.probes;
+
+import static java.time.temporal.ChronoUnit.SECONDS;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Duration;
+import java.util.Optional;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+@Component
+@Order(AbstractOpenIssuesProbe.ORDER)
+class JiraOpenIssuesProbe extends AbstractOpenIssuesProbe {
+    public static final String KEY = "jira-open-issues";
+    private static final String JIRA_HOST = "https://issues.jenkins.io/rest/api/latest/search?";
+    private static final Logger LOGGER = LoggerFactory.getLogger(JiraOpenIssuesProbe.class);
+    final ObjectMapper objectMapper = new ObjectMapper();
+    HttpClient httpClient;
+    HttpRequest httpRequest;
+
+    /**
+     * Get total number of open JIRA issues in a plugin.
+     *
+     * @param context {@link ProbeContext}
+     * @return Optional of type Integer.
+     */
+    @Override
+    Optional<Integer> getCountOfOpenIssues(ProbeContext context) {
+        String viewJiraIssuesUrl = context.getIssueTrackerUrlsByNames().get("jira");
+
+        if (viewJiraIssuesUrl == null || viewJiraIssuesUrl.isEmpty()) {
+            LOGGER.info("JIRA issues not found in Update Center for the plugin.");
+            return Optional.empty();
+        }
+        try {
+            /* The `url` will contain the JIRA url to view issues.
+               For ex: https://issues.jenkins.io/rest/api/latest/search?jql=component=15979
+            */
+            URL url = new URL(viewJiraIssuesUrl);
+
+            /* Here, the query of the url "?jql=component=1833" is concatenated with " AND status=open".
+               This gives the final API required to fetch JIRA issues.
+               For ex: https://issues.jenkins.io/rest/api/latest/search?jql=component=15979%20and%20status=open
+             */
+            String api = JIRA_HOST.concat(url.getQuery())
+                .concat("%20AND%20")
+                .concat("status=open");
+
+            httpRequest = HttpRequest.newBuilder()
+                .uri(new URI(api))
+                .timeout(Duration.of(5, SECONDS))
```

Comment:
Isn't this too quick? What is the default value?

IMHO, for any decision like this, it is interesting that you leave a simple comment in the source code explaining why you took that decision, so readers can understand if it is a deliberate choice, and what is the rationale for it.

General remark: remember you mostly write code for other humans, so you need to tell them your intention.

Reply:
The [JavaDoc ](https://docs.oracle.com/en/java/javase/11/docs/api/java.net.http/java/net/http/HttpRequest.Builder.html#timeout(java.time.Duration))says:

>  The effect of not setting a timeout is the same as setting an infinite Duration, i.e. block forever.

I was not really sure which value to pick. Honestly, the APIs I had tested took less than a second to respond. So I figured 5 seconds is apt.

What should I actually write as the comment?

Reply:
> What should I actually write as the comment?

The general rule to follow while writing source code is that someone reading it should not have to guess why you did certain things, but should find the information written explicitely.

If you thought that 5 seconds was a good timeout value based on your testing, then write it in a simple comment, so it is now our shared understanding.

Basically, imagine reading that piece of source code if you did not write it yourself, and try to see if you understand everything from what you read.